lib/rules/no-shadow.js

```diff
@@ -75,10 +109,176 @@ module.exports = {
+
+			return variable.isValueVariable !== isShadowedValue;
+		}
+
+		/**
+		 * Checks if it's a function type parameter shadow
+		 * @param {Object} variable The variable to check
+		 * @returns {boolean} Whether it's a function type parameter shadow case to ignore
+		 */
+		function isFunctionTypeParameterNameValueShadow(variable) {
+			if (ignoreFunctionTypeParameterNameValueShadow !== true) {
+				return false;
+			}
+
+			return variable.defs.every(def =>
+				ALLOWED_FUNCTION_VARIABLE_DEF_TYPES.has(def.node.type),
+			);
+		}
+
+		/**
+		 * Checks if the variable is a generic of a static method
+		 * @param {Object} variable The variable to check
+		 * @returns {boolean} Whether the variable is a generic of a static method
+		 */
+		function isGenericOfStaticMethod(variable) {
+			if (!("isTypeVariable" in variable)) {
+				return false;
+			}
+
+			if (!variable.isTypeVariable) {
+				return false;
+			}
+
+			if (variable.identifiers.length === 0) {
+				return false;
+			}
+
+			const typeParameter = variable.identifiers[0].parent;
+			if (typeParameter.type !== "TSTypeParameter") {
+				return false;
+			}
+			const typeParameterDecl = typeParameter.parent;
+			if (typeParameterDecl.type !== "TSTypeParameterDeclaration") {
+				return false;
+			}
+			const functionExpr = typeParameterDecl.parent;
+			if (
+				functionExpr.type !== "FunctionExpression" &&
+				functionExpr.type !== "TSEmptyBodyFunctionExpression"
+			) {
+				return false;
+			}
+			const methodDefinition = functionExpr.parent;
+			if (methodDefinition.type !== "MethodDefinition") {
+				return false;
+			}
+			return methodDefinition.static;
+		}
+
+		/**
+		 * Checks if the variable is a generic of a class
+		 * @param {Object} variable The variable to check
+		 * @returns {boolean} Whether the variable is a generic of a class
+		 */
+		function isGenericOfClass(variable) {
+			if (!("isTypeVariable" in variable)) {
+				return false;
+			}
+
+			if (!variable.isTypeVariable) {
+				return false;
+			}
+
+			if (variable.identifiers.length === 0) {
+				return false;
+			}
+
+			const typeParameter = variable.identifiers[0].parent;
+			if (typeParameter.type !== "TSTypeParameter") {
+				return false;
+			}
+			const typeParameterDecl = typeParameter.parent;
+			if (typeParameterDecl.type !== "TSTypeParameterDeclaration") {
+				return false;
+			}
+			const classDecl = typeParameterDecl.parent;
+			return (
+				classDecl.type === "ClassDeclaration" ||
+				classDecl.type === "ClassExpression"
+			);
+		}
+
+		/**
+		 * Checks for static method generic shadowing class generic
+		 * @param {Object} variable The variable to check
+		 * @param {Object} shadowedVariable The shadowed variable
+		 * @returns {boolean} Whether it's a static method generic shadowing class generic
+		 */
+		function isGenericOfAStaticMethodShadow(variable, shadowedVariable) {
+			return (
+				isGenericOfStaticMethod(variable) &&
+				isGenericOfClass(shadowedVariable)
```

Comment:
[Refactor] Interesting: removing the `isGenericOfClass` check here doesn't fail any unit tests. I believe that's because if the variable is of a static method, it must also be of a class. Nothing else would receive `static: true`.

```suggestion
				isGenericOfStaticMethod(variable)
```